rclcpp/include/rclcpp/node.hpp

```diff
@@ -1593,6 +1593,9 @@ class Node : public std::enable_shared_from_this<Node>
   const rclcpp::NodeOptions node_options_;
   const std::string sub_namespace_;
   const std::string effective_namespace_;
+
+  class NodeImpl;
+  std::shared_ptr<NodeImpl> hidden_impl_;
```

Comment:
Could/should this be a `unique_ptr`?

Reply:
No because of the "sub-node" protected constructor, which is a copy constructor that wants to share all the data

Reply:
I'm resolving this as I think we agreed that this should be a `shared_ptr`.